### PR TITLE
fix: make stepper region focusable if scrollable

### PR DESCRIPTION
**Pull request by cpatil-c-eightfold**
## SUMMARY:
In the case of onboarding flow, was getting axe devtools issue saying "Scrollable region must have keyboard access". So added a tabindex attribute to make the stepper region focusable to fix the issue.

## JIRA TASK (Eightfold Employees Only):
https://eightfoldai.atlassian.net/browse/ENG-85455

## CHANGE TYPE:

- [x] Bugfix Pull Request
- [ ] Feature Pull Request

## TEST COVERAGE:

- [x] Tests for this change already exist
- [ ] I have added unittests for this change

## TEST PLAN:
image

- pull this locally and bring up storybook
- confirm that the stepper center region is focusable and user can scroll using the keyboard or not.
- confirm that other variants of the stepper component are still working as expected too.

### CODE REVIEW

File: src/components/Stepper/Stepper.tsx [hunk 1]
```diff
@@ -522,6 +522,7 @@ export const Stepper: FC<StepperProps> = React.forwardRef(
                   className={styles.stepsContainer}
                   onScroll={handleScroll}
                   ref={stepsContainerRef}
+                  tabIndex={0}
```
Comment (jhoward-eightfold) [inline src/components/Stepper/Stepper.tsx:525]: seems like you could do this dynamically such that it isn't in the tab order when the scroll isn't present. something like (please test)

```suggestion
                  tabIndex={mergedScrollable ? 0 : null}
```

File: src/components/Stepper/Stepper.tsx
Comment (jhoward-eightfold) [inline src/components/Stepper/Stepper.tsx:1]: please add a test case that covers this scenario.

File: src/components/Stepper/Stepper.tsx (hunk 1 above)
Comment (cpatil-c-eightfold) [reply src/components/Stepper/Stepper.tsx]: Updated tabindex attribute condition as suggested.

File: src/components/Stepper/Stepper.tsx
Comment (cpatil-c-eightfold) [reply src/components/Stepper/Stepper.tsx:1]: added a test case for scrollable stepper

**Review by jhoward-eightfold — APPROVED** [review]

**Pull request merged**